### ISSUE TITLE
[bug] PST Grimoire of Pestilent Thought is not able to have its evil way

### ISSUE THREAD

**Issue by chilvence** [gemrb/gemrb#600]
Who doesn't like reading a book that convinces you to do evil things like killing your friends, family, pets etc for money and power, I know I can't put those down

In PST if you decide to kill or pickpocket Mantuok the were-rat you find he has a copy of my favourite book the Grimoire of Pestilent Thought. It tells you how to sell your friends into slavery, how to kill them for no reason, and also how to sing out of tune, leave the toilet seat up, stroke a cat the wrong way,  etc etc

If you read through the grimoire enough, it tells you it will give you great power if you kill one of your companions, without using magic though, fair dinkum, why not give it a go ey?

The problem that Gemrb has is that one of the conditions in the dialogue file (DPESTAL.DLG) <[note, NOT dpestil.dlg ] relies on checking global variable "PM_KILLED" == 1 which is not set by any in game script. 

Only by searching through the original exe file using IDA is any mention of this variable to be found.

The in game text tells you to kill one of your companions without using magic and presumably this request gave one of the game script programmers such a headache that they asked one of the game engine programmers to step in and help.

 My best guess is that somewhere deep in the original combat  code is a check to see if TNO melee'd someone on the party roster to death. That is really just a guess though, in reality I could care less and would rather see this implemented any way that makes any sort of sense and doesn't upset the rest of the other games in any other way.

Ida 7 is also apparently freeware for non commercial use now, so that's cool.

**Comment by lynxlynxlynx** [comment]
Can you test if it is indeed enough for only the killing blow to be from weapons?

**Comment by bradallred** [comment]
Thanks for the tip on IDA. I've always avoided it before. Somehow missed that news.

**Comment by chilvence** [comment]
I thought there must have been some sort of catch that made it not so great news. Baby steps though?

As far as the game itself is concerned, I went back and did this again using nothing but magic missile spells. I was able to 'achieve' success all the same, the grimoire asks if I have managed to kill a companion without using magic, me knowing that I have made a point to only use magic to kill a 
party member was able to reply 'yes, of course I didn't use any magic'. 

It is pretty interesting how the power of suggestion seems to work when all else fails. Obviously I have zero immunity to it because I was convinced there must be some level of sophistication that isn't actually there.

I was also able to sell Dak'kon to slavery, save the game, reload, ask him to rejoin my party and move on as if nothing had happened using Torment.exe

There isn't really a high bar here. The whole think is wonky. I am quite happy to continue pushing buttons here, but please stop making me hurt Dak'kon, hasn't he been through enough ? :)

**Comment by lynxlynxlynx** [comment]
PM_KILLED likely stands for party-member-killed, so I bet they only checked that, not the actual damage. Should be trivial to do in Actor::Die or Actor::CheckOnDeath, just making sure LastHitter is in the party. Probably best to add a check, so it's only set once you're in this area.

**Comment by chilvence** [comment]
It doesn't necessarily have to be this area where you kill the party member, it's just the previous demand the book makes that forces you to this shop, and the actors there don't blink when you start fighting in front of them so I didn't bother finding a quiet dark alley.

**Comment by lynxlynxlynx** [comment]
What I meant was that it probably matters if we always set this var or not. If you kill a party member before getting the second quest, is it good enough or do you then have to kill another?

**Comment by chilvence** [comment]
Buena pregunta. 

I just tested that theory. I killed Annah before handing over Dak'kon to Vrishika.  

The book dialogue then gets stuck because I have no more party members left to kill off, so it can't ask me who I want to betray and exits out by default response. Then I can talk to the book again, it asks me if I have done what it asked, I say yes, I get the last spell, even though it hasn't even told me what I am supposed to do. So no, it apparently the game makes no checks to see if this quest is at the appropriate stage of completion. 

👍
